File: testapp/src/main/java/com/nerdscorner/mvplib/testapp/events/attribute/activities/presenter/AttributeMainPresenter.java
package com.nerdscorner.mvplib.testapp.events.attribute.activities.presenter;

import org.greenrobot.eventbus.Subscribe;

import com.nerdscorner.mvplib.events.bus.Bus;
import com.nerdscorner.mvplib.events.presenter.BaseActivityPresenter;
import com.nerdscorner.mvplib.testapp.events.attribute.activities.model.AttributeMainModel;
import com.nerdscorner.mvplib.testapp.events.attribute.activities.model.AttributeMainModel.BackgroundTaskCompletedEvent;
import com.nerdscorner.mvplib.testapp.events.attribute.activities.view.AttributeMainView;
import com.nerdscorner.mvplib.testapp.events.attribute.activities.view.AttributeMainView.ActionClickedEvent;

public class AttributeMainPresenter extends BaseActivityPresenter<AttributeMainView, AttributeMainModel> {

    public AttributeMainPresenter(AttributeMainView view, AttributeMainModel model, Bus bus) {
        super(view, model, bus);
    }

    @Subscribe
    public void onActionClicked(ActionClickedEvent event) {
        getView().setTextValue("Executing background task...");
        getModel().doSomethingInBackground();
    }

    @Subscribe
    public void onBackgroundTaskCompleted(BackgroundTaskCompletedEvent event) {
        getView().setTextValue("Background task completed");
    }
}

File: testapp/src/main/java/com/nerdscorner/mvplib/testapp/events/attribute/activities/presenter/AttributeMainPresenter.kt
package com.nerdscorner.mvplib.testapp.events.attribute.activities.presenter

import com.nerdscorner.mvplib.events.bus.Bus
import com.nerdscorner.mvplib.events.presenter.BaseActivityPresenter
import com.nerdscorner.mvplib.testapp.events.attribute.activities.model.AttributeMainModel
import com.nerdscorner.mvplib.testapp.events.attribute.activities.model.AttributeMainModel.BackgroundTaskCompletedEvent
import com.nerdscorner.mvplib.testapp.events.attribute.activities.view.AttributeMainView
import com.nerdscorner.mvplib.testapp.events.attribute.activities.view.AttributeMainView.ActionClickedEvent
import org.greenrobot.eventbus.Subscribe

class AttributeMainPresenter(view: AttributeMainView, model: AttributeMainModel, bus: Bus) :
        BaseActivityPresenter<AttributeMainView, AttributeMainModel>(view, model, bus) {

    @Subscribe
    fun onActionClicked(event: ActionClickedEvent) {
        view.setTextValue("Executing background task...")
        model.doSomethingInBackground()
    }

    @Subscribe
    fun onBackgroundTaskCompleted(event: BackgroundTaskCompletedEvent) {
        view.setTextValue("Background task completed")
    }

    override fun onResume() {
        super.onResume()
        view.withActivity {
        }
    }
}
